R/buttons.R

```diff
@@ -0,0 +1,292 @@
+#' appearance and re-enables itself.
+#'
+#' @section Manual button reset:
+#' In some advanced use cases, it may be necessary to keep a task button in its
+#' busy state even after the normal reactive processing has completed. Calling
+#' `update_task_button(id, state = "busy")` from the server will opt out of any
+#' currently pending reset for a specific task button. After doing so, the
+#' button can be re-enabled by calling `update_task_button(id, state = "ready")`
+#' after each click's work is complete.
+#'
+#' You can also pass an explicit `auto_reset = FALSE` to `input_task_button()`,
+#' which means that button will _never_ be automatically re-enabled and will
+#' require `update_task_button(id, state = "ready")` to be called each time.
+#'
+#' Note that, as a general rule, Shiny's `update` family of functions do not
+#' take effect at the instant that they are called, but are held until the end
+#' of the current reactive cycle. So if you have many different reactive
+#' calculations and outputs, you don't have to be too careful about when you
+#' call `update_task_button(id, busy = FALSE)`, as the button on the client will
+#' not actually re-enable until the same moment that all of the updated outputs
+#' simultaneously sent to the client.
+#'
+#' @param id The `input` slot that will be used to access the value.
+#' @param label The label of the button while it is in ready (clickable) state;
+#'   usually a string.
+#' @param icon An optional icon to display next to the label while the button is
+#'   in ready state. See [fontawesome::fa_i()].
+#' @param label_busy The label of the button while it is busy.
+#' @param icon_busy The icon to display while the button is busy. By default,
+#'   `fontawesome::fa_i("refresh", class = "fa-spin", "aria-hidden" = "true")`
+#'   is used, which displays a spinning "chasing arrows" icon. You can create
+#'   spinning icons out of other Font Awesome icons by using the same
+#'   expression, but replacing `"refresh"` with a different icon name. See
+#'   [fontawesome::fa_i()].
+#' @param type One of the Bootstrap theme colors (`"primary"`, `"default"`,
+#'   `"secondary"`, `"success"`, `"danger"`, `"warning"`, `"info"`, `"light"`,
+#'   `"dark"`), or `NULL` to leave off the Bootstrap-specific button CSS classes
+#'   altogether.
+#' @param ... Named arguments become attributes to include on the `<button>`
+#'   element.
+#' @param auto_reset If `TRUE` (the default), automatically put the button back
+#'   in "ready" state after its click is handled by the server.
+#'
+#' @section Server value:
+#' An integer of class `"shinyActionButtonValue"`. This class differs from
+#' ordinary integers in that a value of 0 is considered "falsy".
+#' This implies two things:
+#'   * Event handlers (e.g., [shiny::observeEvent()], [shiny::eventReactive()]) won't execute on initial load.
+#'   * Input validation (e.g., [shiny::req()], [shiny::need()]) will fail on initial load.
+#'
+#' @examples
```

Comment:
IIRC there are several (pkgdown) benefits to this

```suggestion
#' @examplesIf interactive()
```

Reply:
Oh awesome